### PR TITLE
chore: Convert `db` unit tests to node:test

### PR DESCRIPTION
**Pull request by amychisholm03**

### CODE REVIEW

File: test/unit/db/trace.test.js [hunk 1]
```diff
@@ -5,13 +5,13 @@
 
 'use strict'
 
-const tap = require('tap')
```
Comment (bizob2828) [inline test/unit/db/trace.test.js:8]: you have some tests that are callback based in here. they have to pass the `end()` cb.  it's the 2nd arg to tests

**Review by bizob2828 — CHANGES_REQUESTED** [review]
both trace.test.js and query-trace-aggregator.test.js have async tests that call functions that execute calls backs, you have to replace the instances of `t.end()` with `end()` which is the 2nd arg to the t.test

File: test/unit/db/trace.test.js [hunk 2]
```diff
@@ -5,13 +5,13 @@
...
+const test = require('node:test')
+const assert = require('node:assert')
 const helper = require('../../lib/agent_helper')
 
-tap.test('SQL trace attributes', function (t) {
-  t.autoend()
+test('SQL trace attributes', async (t) => {
   t.beforeEach(function (t) {
-    t.context.agent = helper.loadMockedAgent({
+    t.mock.agent = helper.loadMockedAgent({
```
Comment (jsumners-nr) [inline test/unit/db/trace.test.js:14]: Tap provided a specific `context` namespace on the test context (`t`) for us to store test scoped data. Unfortunately, `node:test` only provides a test scoped context. So we need to add our own namespace. I have been writing tests like:

```js
test('some feature', async t => {
	t.beforeEach(ctx => {
		ctx.nr = {}
		ctx.nr.agent = {agent: 'instance'}
	})

	t.afterEach(ctx => {
		unload(ctx.nr.agent)
	})

	await t.test('one'), (t, end) => {
		const { agent } = t.nr
		agent.someMethod((error) => {
			assert.equal(error, undefined)
			end()
		})
	})
})
```

What you have written in this test is adding to the test's scoped `mock` tracker entity (see https://nodejs.org/api/test.html#class-mocktracker). Which means we would be adding things to a namespace we don't control and could possibly encounter naming collisions.

In short, instead of attaching things to `t.mock`, please attach them to `t.nr`.

**Review by jsumners-nr — CHANGES_REQUESTED** [review]

**Review by jsumners-nr — DISMISSED** [review]
Fabulous work.

File: test/unit/db/query-parsers/sql.test.js [hunk 3]
```diff
@@ -158,7 +158,7 @@ test('database query parser', async (t) => {
           t.todo('should strip database names from collection names as ' + cat.table)
         } else {
           await t.test('should parse the collection as ' + cat.table, function () {
```
Comment (bizob2828) [inline test/unit/db/query-parsers/sql.test.js:160]: can you remove this sub test, it's not necessary, my suggestion got messed up it appears

File: test/unit/db/query-parsers/sql.test.js [hunk 4]
```diff
@@ -66,121 +62,102 @@ test('database query parser', function (t) {
...
+      assert.equal(ps.query, '')
     })
   })
 
-  t.test('CAT', function (t) {
-    t.autoend()
-    CATs.forEach(function (cat) {
-      t.test(clean(cat.input), function (t) {
-        t.autoend()
+  await t.test('CAT', async function (t) {
```
Comment (bizob2828) [inline test/unit/db/query-parsers/sql.test.js:146]: something is getting missed during the back and forth. this is what the test should look like

```js
  await t.test('CAT', async function (t) {
    for (const cat of CATs) {
      await t.test(clean(cat.input), async (t) => {
        const ps = parseSql(cat.input)

        assert.equal(ps.operation, cat.operation, `should parse the operation as ${cat.operation}`)

        if (cat.table === '(subquery)') {
          t.todo('should parse subquery collections as ' + cat.table)
        } else if (/\w+\.\w+/.test(ps.collection)) {
          t.todo('should strip database names from collection names as ' + cat.table)
        } else {
          assert.equal(ps.collection, cat.table, `should parse the collection as ${cat.table}`)
        }
      })
    }
  })
```

**Review by bizob2828 — APPROVED** [review]
